### PR TITLE
chore: docker compose creation

### PR DESCRIPTION
**Pull request by salet-gutierrez-jalafund**
## Trello Ticket

[ID-19](https://trello.com/c/GC2Bkxyz)

## What did I do?

I added the docker compose so that the application containers can be created, in the same way the Dockerfile was corrected so that the docker compose can be created

## Why did I do this?

I added a file where the necessary logic will be to be able to create a container and so that a database can be added.

## How did I do it?

I did it with the help of docker to create its container.

## Notes:

Before to start run the project, u need to have started the docker service: Then use the following command at the root of the project 
``` 
docker-compose build 
```
 after: 
``` 
docker-compose up --build 
```
 you can check if the containers are up and running with: 
``` 
docker-compose ps 
```

### CODE REVIEW

**Review by alex-paca-jalafund — APPROVED** [review]
Nice work

Screenshot from 2024-09-17 22-24-23

File: docker-compose.yml [hunk 1]
```diff
@@ -0,0 +1,36 @@
...
+  webapp:
+    container_name: expense-tracker-api
+    build:
+      context: .
+      dockerfile: Dockerfile
+    ports: 
+      - "5001:5001" 
+    environment:
+      - ASPNETCORE_ENVIRONMENT=Development
+      - Database__ConnectionString__DefaultConnection=Host=postgres;Port=5432;Database=ExpenseTrackerGrupo2;Username=expenseTrackerGrupoTwo;Password=123456789
```
Comment (joseluis-teran-jalafund) [inline docker-compose.yml:13]: Would you mind to change the name of the Database please? It's not common with upper letters.

Comment (joseluis-teran-jalafund) [inline docker-compose.yml]: Would you mind to change the name of the Database please? It's not common to see it with upper letters.

File: docker-compose.yml [hunk 2]
```diff
@@ -0,0 +1,36 @@
...
+    depends_on:
+      postgres:
+        condition: service_healthy
+
+  postgres:
+    image: postgres:latest
+    container_name: expense-tracker-db
+    environment:
+      POSTGRES_DB: ExpenseTrackerGrupo2
```
Comment (joseluis-teran-jalafund) [inline docker-compose.yml]: Don't forget here also.

File: docker-compose.yml [hunk 3]
```diff
@@ -0,0 +1,36 @@
...
+      POSTGRES_USER: expenseTrackerGrupoTwo
+      POSTGRES_PASSWORD: 123456789
+    ports:
+      - "5432:5432"
+    volumes:
+      - pgdata:/var/lib/postgresql/data
+    healthcheck:
+      test: ["CMD-SHELL", "pg_isready -U expenseTrackerGrupoTwo -d ExpenseTrackerGrupo2"]
```
Comment (joseluis-teran-jalafund) [inline docker-compose.yml]: Same here too.

**Review by joseluis-teran-jalafund — CHANGES_REQUESTED** [review]

File: docker-compose.yml (hunk 1 above)
Comment (salet-gutierrez-jalafund) [reply docker-compose.yml]: done

File: docker-compose.yml (hunk 2 above)
Comment (salet-gutierrez-jalafund) [reply docker-compose.yml]: done

File: docker-compose.yml (hunk 3 above)
Comment (salet-gutierrez-jalafund) [reply docker-compose.yml]: done

**Review by joseluis-teran-jalafund — APPROVED** [review]
Nice.

**Comment by salet-gutierrez-jalafund** [comment]
[ID-19: Project Initialization](https://trello.com/c/GC2Bkxyz/27-id-19-project-initialization)

**Pull request merged**